### ISSUE TITLE
get tablet buttons?

### ISSUE THREAD

**Issue by ZvikaZ** [pyglet/pyglet#611]
Hi.
I'm using Wacom Intuos, and I'm able to get all the pen's behaviour with https://github.com/pyglet/pyglet/blob/master/examples/input/tablet.py.
However, the tablet itself has 4 buttons. How can I be notified of them been pressed?

**Comment by benmoran56** [comment]
I don't have a Tablet handy, but the tablet object should have a `cursors` list. Each cursor should have a `.device` property, which will have a list of `controls`. That should contain a list of all the `Buttons`, which you can then attach event handlers to (or query their `.value`).

**Comment by ZvikaZ** [comment]
These are the cursors that my tablet has:

     print(tablet.cursors)
     [WintabCursor('Puck'), WintabCursor('Pressure Stylus'), WintabCursor('Eraser'), WintabCursor('Puck'), WintabCursor('Pressure Stylus'), WintabCursor('Eraser')]

I don't understand what's this `Puck` about, but beside it, all of them are related to the pen itself.

Anyway, each of these cursor`.device` refers back to `tablet`:

     print([c.device == tablet for c in tablet.cursors])
     [True, True, True, True, True, True]

**Comment by caffeinepills** [comment]
I am looking at the wintab api and I don't really see anything related to buttons.  (https://developer-docs.wacom.com/intuos-cintiq-business-tablets/docs/wintab-basics) However, I do not have any tablets myself, or have messed with them much.

Could you try checking the device list if it has the buttons? `devices = pyglet.input.get_devices()` look in that list for your device and try `device.open()` and if you find it you can do `device.get_controls()`? From there you should be able to get the value or attach an `on_change` event.

**Comment by ZvikaZ** [comment]
Well, I did some investigations, with this snippet:

    import pyglet
    
    def change_handler(args):
        print("handler: ", args)
    
    
    for d in pyglet.input.get_devices():
        if "Wacom" in d.name:
            d.open()
            for control in d.get_controls():
                # if type(control) == pyglet.input.base.Button:
                    print(control)
                    control.set_handler('on_change', change_handler)
    
    pyglet.app.run()

There are indeed a lot of reported buttons and controllers, and even 2 `Watcom` devices. But none did help.

However, as you asked later in your `EDIT`, the question is what are those buttons used for. Currently, I can map them through Watcom settings. I hoped that I can get the "raw input", and use those buttons from my code. However, it seems that these buttons get their assignment in some low level manner. 

For example, currently 2 of my buttons are defined in Watcom settings as "Undo" and "Redo". When I click them, I get the same events in my Pyglet program as `Ctrl+Z` and `Ctrl+Y`.

Bottom line - this probably a non issue. But thanks for helping out with understand it!
